src/rgw/rgw_auth.cc

```diff
@@ -309,6 +309,11 @@ rgw::auth::Strategy::add_engine(const Control ctrl_flag,
 
 
 /* rgw::auth::RemoteAuthApplier */
+bool rgw::auth::RemoteApplier::is_user_type(const RGWUserSourceType& type) const
+{
+  return type == static_cast<RGWUserSourceType>(info.acct_type);
```

Comment:
What about changing the type of `RemoteApplier::AcctInfo::acct_type` from `uint32_t` to `RGWUserSourceType`? This would allow to avoid the clutter of `static_casts`.